app/app.go

```diff
@@ -106,35 +123,44 @@ func (app *App) TuneNote(noteID string) error {
 	if err != nil {
 		return err
 	}
+	/*
+		Do not apply the note, if it was applied before
+		Otherwise, the state file (serialised parameters) will be overwritten, and it will no longer
+		be possible to revert the note to the state before it was tuned.
+	*/
+	_, err = os.Stat(app.State.GetPathToNote(noteID))
+	if err == nil {
+		// state file for note already exists
+		// do not apply the note again
+		log.Printf("note '%s' already applied. Nothing to do", noteID)
+		return nil
+	}
+
 	solNotes := app.GetSortedSolutionEnabledNotes()
 	searchInSol := sort.SearchStrings(solNotes, noteID)
 	searchInNote := sort.SearchStrings(app.TuneForNotes, noteID)
 	if !(searchInSol < len(solNotes) && solNotes[searchInSol] == noteID) && !(searchInNote < len(app.TuneForNotes) && app.TuneForNotes[searchInNote] == noteID) {
 		// Note is not covered by any of the existing solution, hence adding it into the additions' list
 		app.TuneForNotes = append(app.TuneForNotes, noteID)
 		sort.Strings(app.TuneForNotes)
-		if err := app.SaveConfig(); err != nil {
-			return err
-		}
 	}
-	/*
-		Do not apply the note if it was applied before
-		Otherwise, the state file (serialised parameters) will be overwritten, and it will no longer
-		be possible to revert the note to the state before it was tuned.
-	*/
-	_, _, valApplyList, err := app.VerifyNote(noteID)
+	// to prevent double noteIDs in the apply order list
+	i := PositionInNoteApplyOrder(app.NoteApplyOrder, noteID)
+	if i < 0 { // noteID not yet avaialbe
```

Comment:
Just a typo in the comment, `avaialbe` -> `available`

Reply:
Seems I have a problem with the word 'available' :-). Will fix it soon.

Reply:
new commit done